### PR TITLE
Add support for batch invoke endpoint job

### PR DESCRIPTION
**Pull request by nancy-mejia**
# Description

Adding support for batch job endpoint invoke
Spec: https://github.com/Azure/azureml_run_specification/blob/master/specs/job-endpoint.md 

Need to support inputs, outputs, job_name. 
Batch invoke inference should not be affected by the changes.
Changes are under private preview.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by banibrata** [comment]
I don't see any SDK test for this feature unless I am missing something!

**Review by banibrata — CHANGES_REQUESTED** [review]
Please add few tests!

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/batch_job.py [hunk 1]
```diff
@@ -58,6 +59,13 @@ def make(self, data: Any, **kwargs: Any) -> Any:
                         data[EndpointYamlFields.BATCH_JOB_INPUT_DATA][key] = UriFileJobInput(uri=input_data.path)
                     elif input_data.type == AssetTypes.URI_FOLDER:
                         data[EndpointYamlFields.BATCH_JOB_INPUT_DATA][key] = UriFolderJobInput(uri=input_data.path)
+
+        if data.get(EndpointYamlFields.BATCH_JOB_OUTPUT_DATA, None):
+            for key, output_data in data[EndpointYamlFields.BATCH_JOB_OUTPUT_DATA].items():
+                if isinstance(output_data, Output):
+                    if output_data.type == AssetTypes.URI_FILE:
```
Comment (Man-MSFT) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/batch_job.py:66]: Why only uri file? What about uri folders?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py [hunk 2]
```diff
@@ -292,7 +292,8 @@ def create_or_update(
                         **self._init_args,
                     )
 
-        component._set_is_anonymous(kwargs.pop("is_anonymous", False))
+        if not (hasattr(component, "_is_anonymous") and component._is_anonymous is True):
```
Comment (Man-MSFT) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py:295]:
```suggestion
        if not (hasattr(component, "_is_anonymous") and component._is_anonymous):
```

**Review by Man-MSFT — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/batch_job.py (hunk 1 above)
Comment (nancy-mejia) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/batch_job.py:66]: We don't have support to uri_folders in the current swagger version we have. Will create a following PR to include them

File: sdk/ml/azure-ai-ml/tests/batch_services/e2etests/test_batch_endpoint.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+from ast import Call
```
Comment (banibrata) [inline sdk/ml/azure-ai-ml/tests/batch_services/e2etests/test_batch_endpoint.py:1]: Is this being used somewhere?

**Review by banibrata — APPROVED** [review]

**Pull request merged**